PixPanz/byte_le_royale_2021
# working on build updating

Alright, this should fix all the problems I found in the build updater. It took me a while, and I have little experience in this, so double check to make sure if everything here works properly. 

To test:
- Authentication. Since the repo is still private, you will need to create a personal access token on your account to access this private repo through the API. This can be done by going to your settings and creating a personal access token. I dont know what permissions the token will need. Then, create a my_token.py file in the wrapper folder that's modelled after the my_token_example.py file. Your username is your github username. This should allow you authentication. Alternatively, make this repo public and no authentication *should* be required. If so, we're in trouble in other ways and bytele slave will need to get working. 
- Versioning. From here on out, new releases will require the dev to bump the versioning in wrapper/version.py. This is important for the program to determine if a launcher is "newer" or not. If you run ./build.bat with a wrapper/version.py number *lower* than the active release that's out, then running 'python ./launcher.pyz update' will cause the updater to pull the active release from GitHub. 

So run ./build.bat locally with a lower version number ( like 0.0.0 ) and then run python launcher.pyz update. You should successfully update to the latest version. If you get an error like "Not Found", then that means your authentication is incorrect. If it says you are at the latest version, then your wrapper/version.py version number is higher than the latest (0.0.2 as of this PR).

Do *not* mark a release as a pre-release, or this updater will ignore it. The try except blocks are only in place until this repo goes public.
